player.cpp

```diff
@@ -56,21 +56,25 @@ Player Player::parseUnit(const std::string &fileName)
 bool Player::hit(Player *otherPlayer)
 {
     bool fatality;
+    unsigned short damageAmount;
 
     if (otherPlayer->hp <= this->damage)
     {
+        damageAmount = otherPlayer->hp;
         otherPlayer->hp = 0;
         fatality = true;
     }
     else
     {
+        damageAmount = this->damage;
         otherPlayer->hp -= this->damage;
         fatality = false;
     }
 
     this->nextAttack += this->attackCooldown;
+
     // Increase XP
-    this->increaseXP(this->damage);
+    this->increaseXP(damageAmount);
```

Comment:
OK, ez már jó így. A HP csökkentését is ki lehetett volna így emelni, de nem baj. (`otherPlayer->hp -= damageAmount`)